.github/scripts/versioning.py

```diff
@@ -69,27 +69,25 @@ def ask_yes_no_question(question):
     return answer
 
 
-def list_files_in_a_component(component, afr_path, exclude_dirs=[], ext_filter=['.c', '.h']):
+def list_files_in_a_component(component, afr_path, exclude_dirs=[], ext_filter=['.c', '.h'], exclude_hidden=True):
     '''
     Returns a list of all the files in a component.
     '''
     list_of_files = []
     search_path = os.path.join(afr_path, component)
 
     for root, dirs, files in os.walk(search_path, topdown=True):
-        dirs[:] = [d for d in dirs if d not in exclude_dirs]
-
-        # Do not include hidden files and folders.
-        dirs[:] = [d for d in dirs if not d[0] == '.']
-        files = [f for f in files if not f[0] == '.']
+        # Current root is an excluded dir so skip
+        if root in exclude_dirs:
+            continue
 
         for f in files:
-            if ext_filter != None:
-                ext = '.' + f.split('.')[-1]
-                if ext in ext_filter:
+            if exclude_hidden and f[0] == '.':
+                continue
+
+            if (ext_filter is None
```

Comment:
May be worth wrapping the AND expression in a () to emphasis the precedence of the logical operators